desktop/core/src/desktop/js/utils/hooks/UseFileUpload/useChunkUpload.ts

```diff
@@ -0,0 +1,151 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useState } from 'react';
+import useSaveData from '../useSaveData';
+import useQueueProcessor from '../useQueueProcessor';
+import {
+  DEFAULT_CHUNK_SIZE,
+  DEFAULT_CONCURRENT_CHUNK_UPLOAD,
+  FileUploadStatus
+} from '../../constants/storageBrowser';
+import useLoadData from '../useLoadData';
+import { TaskServerResponse, TaskStatus } from '../../../reactComponents/TaskBrowser/TaskBrowser';
+import {
+  createChunks,
+  getChunksCompletePayload,
+  getChunkItemPayload,
+  getChunkSinglePayload,
+  getStatusHashMap,
+  getTotalChunk,
+  UploadChunkItem,
+  UploadItem
+} from './util';
+
+interface UseUploadQueueResponse {
+  addFiles: (item: UploadItem[]) => void;
+  removeFile: (item: UploadItem) => void;
+  isLoading: boolean;
+}
+
+interface ChunkUploadOptions {
+  onStatusUpdate: (item: UploadItem, newStatus: FileUploadStatus) => void;
+  onComplete: () => void;
+}
+
+const useChunkUpload = ({
+  onStatusUpdate,
+  onComplete
+}: ChunkUploadOptions): UseUploadQueueResponse => {
+  const [pendingItems, setPendingItems] = useState<UploadItem[]>([]);
+
+  const { save } = useSaveData(undefined, {
+    postOptions: {
+      qsEncodeData: false,
+      headers: {
+        'Content-Type': 'multipart/form-data'
+      }
+    }
+  });
+
+  const updateItemStatus = (serverResponse: TaskServerResponse[]) => {
+    const statusMap = getStatusHashMap(serverResponse);
+
+    const remainingItems = pendingItems.filter(item => {
+      const status = statusMap[item.uuid];
+      if (status === TaskStatus.Success || status === TaskStatus.Failure) {
+        const ItemStatus = TaskStatus.Success ? FileUploadStatus.Uploaded : FileUploadStatus.Failed;
+        onStatusUpdate(item, ItemStatus);
+        return false;
+      }
+      return true;
+    });
+    if (remainingItems.length === 0) {
+      onComplete();
+    }
+    setPendingItems(remainingItems);
+  };
+
+  const { data: tasksStatus } = useLoadData<TaskServerResponse[]>(
+    '/desktop/api2/taskserver/get_taskserver_tasks/',
+    {
+      pollInterval: pendingItems.length ? 5000 : undefined,
```

Comment:
In current approach:
- Polling only happens when files are successfully uploaded and awaiting final status from the task server.
- Polling stops once all previous files are processed (skip: !pendingItems.length) and resumes when new files finish uploading.

So, in case of large file getting uploaded (lets say 1GB file just started uploading), the pooling will happen till any of the previously uploaded file is awaiting its final status. Once the status is received, the pooling will stop automatically, and as soon as that large file gets uploaded, the pool will start again.

In my option, doing longer frequent poll or incremental poll might create confusion, as there might be few files that are marked as success by task server but we will keep showing its status as `uploading`, till the next poll brings updated status.

Reply:
Ok, thanks for explaining